node/index.js

```diff
@@ -582,6 +582,12 @@ TChannelConnection.prototype.onReaderError = function onReaderError(err) {
         localName: self.channel.hostPort,
         error: err
     });
+
+    // TODO instead of resetting send an error frame first.
```

Comment:
Not necessarily. There are two uses of `resetAll()`

 - If the socket closes or errors then `resetAll()` the connection to clear any pending out / in ops.
 - If something goes wrong in tchannel then `resetAll()` the connection to destroy the socket and clear any pending out / in ops.

These need to be either two seperate methods or we need `options` passed into `resetAll()`